schema/mysql/schema.sql

```diff
@@ -1269,7 +1269,12 @@ CREATE TABLE history (
   flapping_history_id binary(20) DEFAULT NULL COMMENT 'flapping_history.id',
   acknowledgement_history_id binary(20) DEFAULT NULL COMMENT 'acknowledgement_history.id',
 
-  event_type enum('notification','state_change','downtime_start', 'downtime_end','comment_add','comment_remove','flapping_start','flapping_end','ack_set','ack_clear') NOT NULL,
+  -- The enum values are ordered in a way that event_type provides a meaningful sort order for history entries with
+  -- the same event_time. state_change comes first as it can cause many of the other events like trigger downtimes,
+  -- remove acknowledgements and send notifications. Similarly, notification comes last as any other event can result
+  -- in a notification. End events sort before the corresponding start events as any ack/comment/downtime/flapping
+  -- period should last for more than a millisecond, therefore, the old period ends first and then the new one starts.
+  event_type enum('state_change', 'ack_clear', 'comment_remove', 'downtime_end', 'flapping_end', 'ack_set', 'comment_add', 'downtime_start', 'flapping_start', 'notification') NOT NULL,
```

Comment:
I would order as follows: `state_change`, `ack_clear`, `downtime_end`, `flapping_end`, `comment_remove`, `comment_add`, `flapping_start`, `downtime_start`, `ack_set`, `notification`.

Reasoning:

* Comments are informative.
* Flapping is automatic and changes mechanics.
* Downtimes are semi-automatic, require user action (or configuration) and change mechanics.
* Acks are pure user actions and change mechanics.

In short, the ordering indicates whether the event changes Icinga mechanics and whether its origin is Icinga or the user.

Reply:
Only changes the ordering within the start and end events (without mixing the two groups), so basically only changes where I've arbitrarily chosen to sort alphabetically, so no complaints from me.

> Reasoning: [...] In short, the ordering indicates whether the event changes Icinga mechanics and whether its origin is Icinga or the user.

So basically assigns the events some kind of importance (ack being the most important and comment the least), so that an `ack_set` will show up as the most recent entry (besides notifications).